src/mlpack/methods/reinforcement_learning/replay/prioritized_replay.hpp

```diff
@@ -0,0 +1,285 @@
+    }
+  }
+
+  /**
+   * Sample some experience according to their priorities.
+   *
+   * @return The indices to be chosen.
+   */
+  arma::ucolvec SampleProportional()
+  {
+    arma::ucolvec idxes(batchSize);
+    double totalSum = idxSum.Sum(0, (full ? capacity : position));
+    double sumPerRange = totalSum / batchSize;
+    for (size_t bt = 0; bt < batchSize; bt++) {
+      double mass = arma::randu() * sumPerRange + bt * sumPerRange;
+      size_t idx = idxSum.FindPrefixSum(mass);
+      idxes[bt] = idx;
+    }
+    return idxes;
+  }
+
+  /**
+   * Sample some experience according to their priorities.
+   *
+   * @param sampledStates Sampled encoded states.
+   * @param sampledActions Sampled actions.
+   * @param sampledRewards Sampled rewards.
+   * @param sampledNextStates Sampled encoded next states.
+   * @param isTerminal Indicate whether corresponding next state is terminal
+   *        state.
+   * @param sampledIndices Sampled indices.
+   * @param weights Corresponding weight for updating the loss.
+   */
+  void Sample(arma::mat& sampledStates,
+              arma::icolvec& sampledActions,
+              arma::colvec& sampledRewards,
+              arma::mat& sampledNextStates,
+              arma::icolvec& isTerminal)
+  {
+    size_t upperBound = full ? capacity : position;
+
+    sampledIndices = SampleProportional();
+    BetaAnneal();
+
+    sampledStates = states.cols(sampledIndices);
+    sampledActions = actions.elem(sampledIndices);
+    sampledRewards = rewards.elem(sampledIndices);
+    sampledNextStates = nextStates.cols(sampledIndices);
+    isTerminal = this->isTerminal.elem(sampledIndices);
+
+    // Calculate the weights of sampled transitions.
+
+    size_t numSample = full ? capacity : position;
+    weights = arma::rowvec(sampledIndices.n_rows);
+
+    for (size_t i = 0; i < sampledIndices.n_rows; i++)
+    {
+      double p_sample = idxSum.Get(sampledIndices[i]) / idxSum.Sum();
+      weights[i] = pow(numSample * p_sample, -beta);
+    }
+    weights /= weights.max();
+  }
+
+  /**
+   * Update priorities of sampled transitions.
+   *
+   * @param indices The indices of sample to be updated.
+   * @param priorities Their corresponding priorities.
+   */
+  void UpdatePriorities(arma::ucolvec& indices, arma::colvec& priorities)
+  {
+      arma::colvec alphaPri = alpha * priorities;
+      maxPriority = std::max(maxPriority, arma::max(priorities));
+      idxSum.BatchUpdate(indices, alphaPri);
+  }
+
+  /**
+   * Get the number of transitions in the memory.
+   *
+   * @return Actual used memory size.
+   */
+  const size_t& Size()
+  {
+    return full ? capacity : position;
+  }
+
+  /**
+   *  Annealing the beta.
+   */
+  void BetaAnneal()
+  {
+    beta = beta + (1 - initialBeta) * 1.0 / replayBetaIters;
+  }
+
+  void Update(arma::mat target, arma::icolvec sampledActions,
+              arma::mat nextActionValues, arma::mat& gradients)
+  {
+    arma::colvec td_error(target.n_cols);
```

Comment:
Minor style issue; can you use camel casing for all names; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more informations.